### PR TITLE
MB-2608 create domestic origin destination sit additional days pricer

### PR DESCRIPTION
**Pull request by gidjin**
## Description

As part of GHC Rate Engine this PR adds the pricer for Domestic Destination and Origin Additional Days SIT.

## Reviewer Notes

This PR hard codes the Number of days (value is 29) pending completion of https://dp3.atlassian.net/browse/MB-1564
You will need the number of days hard coded value for calculating the expected price. Note the payment service item param for number of days will still say not implemented.

Once MB-1564 is complete we should just need to revert commit  ef4bee4894af4c85893251123bde674a99ab9cbc

## Setup

- First, set up the baseline data: `make db_dev_e2e_populate`
- Start up the server and the client: `make server_run` and `make client_run`
- Go to the TOO interface and approve one of the moves. Make sure the orders number, TAC, and SAC are set in the orders too. Make note of the move code so you can get the MTO ID.
- Create a `create_dofsit.json` file that looks like this:
```  
{
  "body": {
    "moveTaskOrderID": "<insert MTO ID>",
    "mtoShipmentID": "<insert shipment ID>",
    "modelType": "MTOServiceItemOriginSIT",
    "reServiceCode": "DOFSIT",
    "sitPostalCode": "00000",
    "reason": "my address changed",
    "sitEntryDate": "2021-01-13",
    "sitHHGActualOrigin":  {
      "streetAddress1": "20245 4th St",
      "city": "Albuquerque",
      "state": "NM",
      "postalCode": "87117",
      "country": "United States"
    }
  }
}
```
- Send this payload via the prime-api-client: `prime-api-client create-mto-service-item --insecure --filename create_dofsit.json`
- Create a `create_ddfsit.json` file that looks like this:
```  
{
  "body": {
    "moveTaskOrderID": "<insert MTO ID>",
    "mtoShipmentID": "<insert shipment ID>",
    "modelType": "MTOServiceItemDestSIT",
    "reServiceCode": "DDFSIT",
    "timeMilitary1": "1705Z",
    "firstAvailableDeliveryDate1": "2021-01-31",
    "timeMilitary2": "0719Z",
    "firstAvailableDeliveryDate2": "2021-02-03",
    "sitEntryDate": "2021-01-12"
  }
}
```
- Send this payload via the prime-api-client: `prime-api-client create-mto-service-item --insecure --filename create_ddfsit.json`
- You should see the additional day SIT codes for that move now in the TOO interface under the "Move task order" tab.  Approve them both
- Create a `create_payment_request.json` file that looks like this:
```
{
  "body": {
    "isFinal": false,
    "moveTaskOrderID": "<insert MTO ID>",
    "serviceItems": [
      {
        "id": "<insert DOASIT MTO service item>"
      },
      {
        "id": "<insert DDASIT MTO service item>"
      }
    ]
  }
}
```
- Send this payload via the prime-api-client: `prime-api-client create-payment-request --insecure --filename create_payment_request.json`
- Go to the "Payment Requests" tab for the move and verify the price on the payment request you just made.  Compare that to what you would expect based on the [SIT Pricing Notes](https://docs.google.com/document/d/1jRl91aslMnOMt2zS1tt_XT0lD1I-xvsiKOfprAaFrAs/edit)

### Sample payment request json

```json
{
  "eTag": "MjAyMS0wMS0yOVQwOToyNDoyOS44MjI4Nlo=",
  "id": "19a2f96c-8d7b-4072-ba46-fd16c158a0a7",
  "isFinal": false,
  "moveTaskOrderID": "9c7b255c-2981-4bf8-839f-61c7458e2b4d",
  "paymentRequestNumber": "4365-8655-11",
  "paymentServiceItems": [
    {
      "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4wMTA3ODJa",
      "id": "f20a6ec1-dd51-41ca-9a1c-006bad8ea400",
      "mtoServiceItemID": "00a3a6e0-17b8-42ed-9902-003c4902feae",
      "paymentRequestID": "19a2f96c-8d7b-4072-ba46-fd16c158a0a7",
      "paymentServiceItemParams": [
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDoyOS45NDQ2MzZa",
          "id": "58147b5d-11c0-4bef-ad1e-a7063f7e62aa",
          "key": "RequestedPickupDate",
          "origin": "PRIME",
          "paymentServiceItemID": "f20a6ec1-dd51-41ca-9a1c-006bad8ea400",
          "type": "DATE",
          "value": "2021-01-29"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDoyOS45NTEzODVa",
          "id": "183a6441-283d-470b-b133-0f5361e73bd6",
          "key": "NumberDaysSIT",
          "origin": "PRIME",
          "paymentServiceItemID": "f20a6ec1-dd51-41ca-9a1c-006bad8ea400",
          "type": "INTEGER",
          "value": "NOT IMPLEMENTED"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDoyOS45NTY0MTFa",
          "id": "46722f0f-9878-45c2-ac87-a4815f1f8bde",
          "key": "WeightBilledActual",
          "origin": "SYSTEM",
          "paymentServiceItemID": "f20a6ec1-dd51-41ca-9a1c-006bad8ea400",
          "type": "INTEGER",
          "value": "1349"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDoyOS45NjMzMzVa",
          "id": "4b26500f-0162-4ed1-b728-221c09a233a9",
          "key": "WeightActual",
          "origin": "PRIME",
          "paymentServiceItemID": "f20a6ec1-dd51-41ca-9a1c-006bad8ea400",
          "type": "INTEGER",
          "value": "1349"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDoyOS45NzA1MTha",
          "id": "7afbcc1c-300f-456a-80ed-07c23239d856",
          "key": "WeightEstimated",
          "origin": "PRIME",
          "paymentServiceItemID": "f20a6ec1-dd51-41ca-9a1c-006bad8ea400",
          "type": "INTEGER",
          "value": "1400"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDoyOS45ODM2MjFa",
          "id": "a95da692-de5a-4663-899a-c620511100f9",
          "key": "ServiceAreaOrigin",
          "origin": "SYSTEM",
          "paymentServiceItemID": "f20a6ec1-dd51-41ca-9a1c-006bad8ea400",
          "type": "STRING",
          "value": "516"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDoyOS45ODk3NDha",
          "id": "b8b7a357-e363-476a-85ee-6defeeca9bcb",
          "key": "ZipPickupAddress",
          "origin": "PRIME",
          "paymentServiceItemID": "f20a6ec1-dd51-41ca-9a1c-006bad8ea400",
          "type": "STRING",
          "value": "87117"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDoyOS45OTUzODZa",
          "id": "2346fd24-f66a-43f5-bfd8-14f59ce99b42",
          "key": "ContractCode",
          "origin": "SYSTEM",
          "paymentServiceItemID": "f20a6ec1-dd51-41ca-9a1c-006bad8ea400",
          "type": "STRING",
          "value": "TRUSS_TEST"
        }
      ],
      "priceCents": 24755,
      "referenceID": "4365-8655-f20a6ec1",
      "status": "REQUESTED"
    },
    {
      "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4xNTQ4NzJa",
      "id": "573d95e4-7c56-4532-bbe3-7f425791cbae",
      "mtoServiceItemID": "2c8712ba-fb29-40b8-9396-be9b8d774b7e",
      "paymentRequestID": "19a2f96c-8d7b-4072-ba46-fd16c158a0a7",
      "paymentServiceItemParams": [
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4xMTI0NjJa",
          "id": "9c234e18-ec8e-44e8-8952-214367a211b5",
          "key": "RequestedPickupDate",
          "origin": "PRIME",
          "paymentServiceItemID": "573d95e4-7c56-4532-bbe3-7f425791cbae",
          "type": "DATE",
          "value": "2021-01-29"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4xMTc3Nlo=",
          "id": "8d680062-6f27-4665-a34a-f8f5fb78c645",
          "key": "NumberDaysSIT",
          "origin": "PRIME",
          "paymentServiceItemID": "573d95e4-7c56-4532-bbe3-7f425791cbae",
          "type": "INTEGER",
          "value": "NOT IMPLEMENTED"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4xMjIwNzNa",
          "id": "c3ca2c45-6b00-41c2-a82d-aca97d906859",
          "key": "WeightBilledActual",
          "origin": "SYSTEM",
          "paymentServiceItemID": "573d95e4-7c56-4532-bbe3-7f425791cbae",
          "type": "INTEGER",
          "value": "1349"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4xMjYxNDla",
          "id": "1924b80e-7de0-4381-96d0-c16ed088cb6a",
          "key": "WeightActual",
          "origin": "PRIME",
          "paymentServiceItemID": "573d95e4-7c56-4532-bbe3-7f425791cbae",
          "type": "INTEGER",
          "value": "1349"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4xMzA0Njla",
          "id": "0b2182b5-d050-42f3-972b-6a5418478f0e",
          "key": "WeightEstimated",
          "origin": "PRIME",
          "paymentServiceItemID": "573d95e4-7c56-4532-bbe3-7f425791cbae",
          "type": "INTEGER",
          "value": "1400"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4xMzY2NTda",
          "id": "a7c0eec1-b16d-4efa-b91a-aa0f0e8ffe51",
          "key": "ServiceAreaDest",
          "origin": "SYSTEM",
          "paymentServiceItemID": "573d95e4-7c56-4532-bbe3-7f425791cbae",
          "type": "STRING",
          "value": "080"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4xNDAwNzla",
          "id": "c0730e76-b6ac-4bbc-8895-d902ad34936c",
          "key": "ZipDestAddress",
          "origin": "PRIME",
          "paymentServiceItemID": "573d95e4-7c56-4532-bbe3-7f425791cbae",
          "type": "STRING",
          "value": "94535"
        },
        {
          "eTag": "MjAyMS0wMS0yOVQwOToyNDozMC4xNDQyMTla",
          "id": "d2304388-8d59-4714-ab66-ae9ed515aa58",
          "key": "ContractCode",
          "origin": "SYSTEM",
          "paymentServiceItemID": "573d95e4-7c56-4532-bbe3-7f425791cbae",
          "type": "STRING",
          "value": "TRUSS_TEST"
        }
      ],
      "priceCents": 21960,
      "referenceID": "4365-8655-573d95e4",
      "status": "REQUESTED"
    }
  ],
  "status": "PENDING"
}
```

## Acceptance Tests

Basically the same as above but on staging instead.

## Code Review Verification Steps

* Any new migrations/schema changes:
  * [X] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [X] Have been communicated to #g-database
* [X] Request review from a member of a different team.
* [X] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-2608) for this change

### CODE REVIEW

**Review by pearl-truss — COMMENTED** [review]
I'm able to make a payment request with the service items, however the EDI generation fails for the payment request. I ran into something similar with DDFSIT where I needed to update the edi generation case statement https://github.com/transcom/mymove/pull/5758/files#diff-cd7fccf038f1ebd436ba86a5c2906b2af226d010e08b551df7f16b4270595480R631

**Comment by gidjin** [comment]
> I'm able to make a payment request with the service items, however the EDI generation fails for the payment request. I ran into something similar with DDFSIT where I needed to update the edi generation case statement https://github.com/transcom/mymove/pull/5758/files#diff-cd7fccf038f1ebd436ba86a5c2906b2af226d010e08b551df7f16b4270595480R631

Good catch, updated accordingly

**Review by pearl-truss — APPROVED** [review]
Changes look 👍

**Review by reggieriser — APPROVED** [review]
Migration looks good.  ContractCode param was not there before I ran it, then appears for both service items after running it.  Note that I haven't looked at the rest of the code, so my approval is for the migration only.
